### DISCUSSION TITLE
Where to configure --strict --strict-builtin-errors and similar options?

### DISCUSSION THREAD

**Discussion by leefernandes** [open-policy-agent/community#560] [Envoy] [answered]
I can't find any documentation on how/where to specify these options for an OPA Envoy agent.

**Comment by ashutosh-narkar** [comment]
` --strict` and `--strict-builtin-errors` are command line flags for the `opa eval` command.  You should be able to use the same commands with OPA Envoy plugin too. What is your use-case?

**Reply by ashutosh-narkar** [reply]
`opa check` also has the `strict` flag.

**Comment by anderseknert** [comment]
`--strict` is a compiler option, so that's probably better to run when you check / package your policies, as @ashutosh-narkar suggests.
`--strict-builtin-errors` is however a runtime option, and assuming the Envoy plugin accepts the same [query parameters](https://www.openpolicyagent.org/docs/latest/rest-api/#query-parameters-3) as OPA does, you should be able to set `strict-builtin-errors` on any request.

**Reply by ashutosh-narkar** [reply, marked as answer]
Enabling strict builtin errors during runtime for OPA-Envoy plugin are not supported atm afair. One way to achieve that would be to extend the plugin config to support that.

**Reply by anderseknert** [reply]
Ah, alright! Thanks for making that clear @ashutosh-narkar
